nncf/torch/quantization/algo.py

```diff
@@ -335,7 +335,10 @@ def generate_setup(self) -> SingleConfigQuantizerSetup:
         if self._debug_interface:
             self._debug_interface.visualize_insertion_point_graph(insertion_point_graph)
         from nncf.common.quantization.quantizer_propagation.solver import QuantizerPropagationSolver
-        prop_graph_solver = QuantizerPropagationSolver(ignored_scopes=self.ignored_scopes,
+        ignored_scopes_for_solver = (lambda x: [x] if isinstance(x, str) else x)(self.ignored_scopes) + \
```

Comment:
Dangerous - it isn't obvious here that we are actually aiming to get either a list of `str` when starting from either a `str` or already a `list` of `str`. I suggest you make a separate named function for this, with asserts/runtime errors for the situation when the starting object is neither a `str` nor a `list` of `str` in which case we can do nothing but crash.

Reply:
A separate function was made but there is no asserts/runtime errors for situation when the starting object is neither a `str` nor a `list` of `str` because this function is used for ignored_scopes processing only which according to json schema 
can be only `str`, `List[str]` or `None` (this also valid for `target_scopes`)
`"ignored_scopes": with_attributes(make_object_or_array_of_objects_schema(_STRING), description=IGNORED_SCOPES_DESCRIPTION),
  "target_scopes": with_attributes(make_object_or_array_of_objects_schema(_STRING), description=TARGET_SCOPES_DESCRIPTION),`
Also None value processing in that particular case can have different logic from processing `None` value for `target_scopes` so this function isn't generalized and implemented as nested
@vshampor, @ljaljushkin  what do you think?

Reply:
I'm ok with this function